### PR TITLE
hyprlock: 0.4.1 -> 0.5.0; sdbus-cpp_2: init at 2.0.0

### PR DESCRIPTION
**Pull request by iynaix**
Diff: https://github.com/hyprwm/hyprlock/compare/v0.4.1...v0.5.0
Changelog: https://github.com/hyprwm/hyprlock/releases/tag/v0.5.0

## Things done

Now requires sdbus-cpp >= 2.0.0, but bumping sdbus-cpp to 2.0 breaks the following other packages:

> dnf5 dnf5.man gummy hypridle jami xdg-desktop-portal-hyprland

I'm not sure if the override is the correct approach to resolve this.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by JohnRTitor — COMMENTED** [review]
Let's init a sdbus-cpp_2 with this PR. 

Dependant packages which can work with v2, should be updated to use this one.

**Comment by iynaix** [comment]
Added sdbus-cpp_2 as a separate commit, updated hyprlock to use the new sdbus-cpp_2 package

**Comment by JohnRTitor** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 350391`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>hyprlock</li>
    <li>sdbus-cpp_2</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 2 packages built:</summary>
  <ul>
    <li>hyprlock</li>
    <li>sdbus-cpp_2</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Review by JohnRTitor — APPROVED** [review]

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1906,7 +1906,8 @@ with pkgs;
 
   scarab = callPackage ../tools/games/scarab { };
 
-  sdbus-cpp = callPackage ../development/libraries/sdbus-cpp { };
+  sdbus-cpp = (callPackage ../development/libraries/sdbus-cpp { }).sdbus-cpp;
+  sdbus-cpp_2 = (callPackage ../development/libraries/sdbus-cpp { }).sdbus-cpp_2;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:1910]:
```suggestion
  inherit (callPackage ../development/libraries/sdbus-cpp { }) sdbus-cpp sdbus-cpp_2;
```

File: pkgs/development/libraries/sdbus-cpp/default.nix [hunk 2]
```diff
@@ -1,53 +1,73 @@
...
+        maintainers = [ ];
+        platforms = lib.platforms.linux;
+        mainProgram = "sdbus-c++-xml2cpp";
+      };
+    };
+in
+{
+  sdbus-cpp = generic {
+    version = "1.5.0";
+    rev = "v1.5.0";
```
[the remark this answers is not in the document]
Comment (JohnRTitor) [reply pkgs/development/libraries/sdbus-cpp/default.nix]: Done, changed some nitpicks as well

**Pull request merged**